resource.go

```diff
@@ -44,6 +46,7 @@ type Config struct {
 	//
 	RESTClient               rest.Interface
 	VersionBundleVersionFunc func(v interface{}) (string, error)
+	Watcher                  func(opts metav1.ListOptions) (watch.Interface, error)
```

Comment:
I tried to list CRs using the rest client but it is not good enough so we need something else. I went with the watcher because its interface is kind of abstract.